This patch has been submitted upstream BZ2681 by Samuel Thibault,
who works on accessibility issues.  It adds a new BRF encoding,
aka ASCII Braille.

Index: iconvdata/Makefile
===================================================================
--- iconvdata/Makefile.orig
+++ iconvdata/Makefile
@@ -51,7 +51,8 @@
 	   GB18030 ISO-2022-CN-EXT VISCII GBBIG5 CP10007 KOI8-T		 \
 	   GEORGIAN-PS GEORGIAN-ACADEMY ISO-IR-209 MAC-SAMI ARMSCII-8	 \
 	   TCVN5712-1 libJISX0213 EUC-JISX0213 SHIFT_JISX0213		 \
-	   ISO-2022-JP-3 TSCII IBM866NAV CP932 EUC-JP-MS PT154 RK1048
+	   ISO-2022-JP-3 TSCII IBM866NAV CP932 EUC-JP-MS PT154 RK1048	 \
+	   BRF
 
 modules.so := $(addsuffix .so, $(modules))
 
@@ -172,7 +173,8 @@
 	      mac-sami.c ibm1160.c ibm1160.h ibm1161.c ibm1161.h	    \
 	      ibm1163.c ibm1163.h ibm1164.c ibm1164.h jisx0213.c jisx0213.h \
 	      euc-jisx0213.c shift_jisx0213.c iso-2022-jp-3.c		    \
-	      tcvn5712-1.c armscii-8.c tscii.c ibm866nav.c pt154.c rk1048.c
+	      tcvn5712-1.c armscii-8.c tscii.c ibm866nav.c pt154.c rk1048.c \
+	      brf.c
 
 # We build the transformation modules only when we build shared libs.
 ifeq (yes,$(build-shared))
@@ -213,7 +215,7 @@
 			iso8859-13 iso8859-14 iso8859-15 mac-uk sami-ws2   \
 			iso-ir-197 tis-620 koi8-u ibm874 cp10007 koi8-t	   \
 			georgian-ps georgian-academy iso-ir-209 mac-sami   \
-			iso8859-11 ibm866nav pt154 rk1048
+			iso8859-11 ibm866nav pt154 rk1048 brf
 
 gen-special-modules := iso8859-7jp
 
Index: iconvdata/brf.c
===================================================================
--- /dev/null
+++ iconvdata/brf.c
@@ -0,0 +1,29 @@
+/* Conversion from and to BRF.
+   Copyright (C) 2006 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+   Contributed by Samuel Thibault <samuel.thibault@ens-lyon.org>, 2006.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+#include <stdint.h>
+
+/* Get the conversion table.  */
+#define TABLES <brf.h>
+
+#define CHARSET_NAME	"BRF//"
+#define HAS_HOLES	1	/* Not all 256 character are defined.  */
+
+#include <8bit-gap.c>
Index: iconvdata/gconv-modules
===================================================================
--- iconvdata/gconv-modules.orig
+++ iconvdata/gconv-modules
@@ -1572,3 +1572,7 @@
 alias	STRK1048-2002//		RK1048//
 module	RK1048//		INTERNAL		RK1048		1
 module	INTERNAL		RK1048//		RK1048		1
+
+#	from			to			module		cost
+module	BRF//			INTERNAL		BRF		1
+module	INTERNAL		BRF//			BRF		1
Index: iconvdata/testdata/BRF
===================================================================
--- /dev/null
+++ iconvdata/testdata/BRF
@@ -0,0 +1,7 @@
+A B C D E F G H I J
+K L M N O P Q R S T
+U V X Y Z & = ( ! )
+* < % ? : $ ] \ [ W
+1 2 3 4 5 6 7 8 9 0
+/ + # > ' -
+@ ^ _ " . ; ,
Index: iconvdata/testdata/BRF..UTF-8
===================================================================
--- /dev/null
+++ iconvdata/testdata/BRF..UTF-8
@@ -0,0 +1,7 @@
+⠁⠀⠃⠀⠉⠀⠙⠀⠑⠀⠋⠀⠛⠀⠓⠀⠊⠀⠚
+⠅⠀⠇⠀⠍⠀⠝⠀⠕⠀⠏⠀⠟⠀⠗⠀⠎⠀⠞
+⠥⠀⠧⠀⠭⠀⠽⠀⠵⠀⠯⠀⠿⠀⠷⠀⠮⠀⠾
+⠡⠀⠣⠀⠩⠀⠹⠀⠱⠀⠫⠀⠻⠀⠳⠀⠪⠀⠺
+⠂⠀⠆⠀⠒⠀⠲⠀⠢⠀⠖⠀⠶⠀⠦⠀⠔⠀⠴
+⠌⠀⠬⠀⠼⠀⠜⠀⠄⠀⠤
+⠈⠀⠘⠀⠸⠀⠐⠀⠨⠀⠰⠀⠠
Index: localedata/charmaps/BRF
===================================================================
--- /dev/null
+++ localedata/charmaps/BRF
@@ -0,0 +1,108 @@
+<code_set_name> BRF
+<comment_char> %
+<escape_char> /
+% version: 1.0
+%  source: Samuel Thibault <samuel.thibault@ens-lyon.org>
+
+CHARMAP
+<U0000>     /x00         NULL (NUL)
+<U0001>     /x01         START OF HEADING (SOH)
+<U0002>     /x02         START OF TEXT (STX)
+<U0003>     /x03         END OF TEXT (ETX)
+<U0004>     /x04         END OF TRANSMISSION (EOT)
+<U0005>     /x05         ENQUIRY (ENQ)
+<U0006>     /x06         ACKNOWLEDGE (ACK)
+<U0007>     /x07         BELL (BEL)
+<U0008>     /x08         BACKSPACE (BS)
+<U0009>     /x09         CHARACTER TABULATION (HT)
+<U000A>     /x0a         LINE FEED (LF)
+<U000B>     /x0b         LINE TABULATION (VT)
+<U000C>     /x0c         FORM FEED (FF)
+<U000D>     /x0d         CARRIAGE RETURN (CR)
+<U000E>     /x0e         SHIFT OUT (SO)
+<U000F>     /x0f         SHIFT IN (SI)
+<U0010>     /x10         DATALINK ESCAPE (DLE)
+<U0011>     /x11         DEVICE CONTROL ONE (DC1)
+<U0012>     /x12         DEVICE CONTROL TWO (DC2)
+<U0013>     /x13         DEVICE CONTROL THREE (DC3)
+<U0014>     /x14         DEVICE CONTROL FOUR (DC4)
+<U0015>     /x15         NEGATIVE ACKNOWLEDGE (NAK)
+<U0016>     /x16         SYNCHRONOUS IDLE (SYN)
+<U0017>     /x17         END OF TRANSMISSION BLOCK (ETB)
+<U0018>     /x18         CANCEL (CAN)
+<U0019>     /x19         END OF MEDIUM (EM)
+<U001A>     /x1a         SUBSTITUTE (SUB)
+<U001B>     /x1b         ESCAPE (ESC)
+<U001C>     /x1c         FILE SEPARATOR (IS4)
+<U001D>     /x1d         GROUP SEPARATOR (IS3)
+<U001E>     /x1e         RECORD SEPARATOR (IS2)
+<U001F>     /x1f         UNIT SEPARATOR (IS1)
+<U2800>     /x20         BRAILLE PATTERN BLANK
+<U282E>     /x21         BRAILLE PATTERN DOTS-2346
+<U2810>     /x22         BRAILLE PATTERN DOTS-5
+<U283C>     /x23         BRAILLE PATTERN DOTS-3456
+<U282B>     /x24         BRAILLE PATTERN DOTS-1246
+<U2829>     /x25         BRAILLE PATTERN DOTS-146
+<U282F>     /x26         BRAILLE PATTERN DOTS-12346
+<U2804>     /x27         BRAILLE PATTERN DOTS-3
+<U2837>     /x28         BRAILLE PATTERN DOTS-12356
+<U283E>     /x29         BRAILLE PATTERN DOTS-23456
+<U2821>     /x2a         BRAILLE PATTERN DOTS-16
+<U282C>     /x2b         BRAILLE PATTERN DOTS-346
+<U2820>     /x2c         BRAILLE PATTERN DOTS-6
+<U2824>     /x2d         BRAILLE PATTERN DOTS-36
+<U2828>     /x2e         BRAILLE PATTERN DOTS-46
+<U280C>     /x2f         BRAILLE PATTERN DOTS-34
+<U2834>     /x30         BRAILLE PATTERN DOTS-356
+<U2802>     /x31         BRAILLE PATTERN DOTS-2
+<U2806>     /x32         BRAILLE PATTERN DOTS-23
+<U2812>     /x33         BRAILLE PATTERN DOTS-25
+<U2832>     /x34         BRAILLE PATTERN DOTS-256
+<U2822>     /x35         BRAILLE PATTERN DOTS-26
+<U2816>     /x36         BRAILLE PATTERN DOTS-235
+<U2836>     /x37         BRAILLE PATTERN DOTS-2356
+<U2826>     /x38         BRAILLE PATTERN DOTS-236
+<U2814>     /x39         BRAILLE PATTERN DOTS-35
+<U2831>     /x3a         BRAILLE PATTERN DOTS-156
+<U2830>     /x3b         BRAILLE PATTERN DOTS-56
+<U2823>     /x3c         BRAILLE PATTERN DOTS-126
+<U283F>     /x3d         BRAILLE PATTERN DOTS-123456
+<U281C>     /x3e         BRAILLE PATTERN DOTS-345
+<U2839>     /x3f         BRAILLE PATTERN DOTS-1456
+<U2808>     /x40         BRAILLE PATTERN DOTS-4
+<U2801>     /x41         BRAILLE PATTERN DOTS-1
+<U2803>     /x42         BRAILLE PATTERN DOTS-12
+<U2809>     /x43         BRAILLE PATTERN DOTS-14
+<U2819>     /x44         BRAILLE PATTERN DOTS-145
+<U2811>     /x45         BRAILLE PATTERN DOTS-15
+<U280B>     /x46         BRAILLE PATTERN DOTS-124
+<U281B>     /x47         BRAILLE PATTERN DOTS-1245
+<U2813>     /x48         BRAILLE PATTERN DOTS-125
+<U280A>     /x49         BRAILLE PATTERN DOTS-24
+<U281A>     /x4a         BRAILLE PATTERN DOTS-245
+<U2805>     /x4b         BRAILLE PATTERN DOTS-13
+<U2807>     /x4c         BRAILLE PATTERN DOTS-123
+<U280D>     /x4d         BRAILLE PATTERN DOTS-134
+<U281D>     /x4e         BRAILLE PATTERN DOTS-1345
+<U2815>     /x4f         BRAILLE PATTERN DOTS-135
+<U280F>     /x50         BRAILLE PATTERN DOTS-1234
+<U281F>     /x51         BRAILLE PATTERN DOTS-12345
+<U2817>     /x52         BRAILLE PATTERN DOTS-1235
+<U280E>     /x53         BRAILLE PATTERN DOTS-234
+<U281E>     /x54         BRAILLE PATTERN DOTS-2345
+<U2825>     /x55         BRAILLE PATTERN DOTS-136
+<U2827>     /x56         BRAILLE PATTERN DOTS-1236
+<U283A>     /x57         BRAILLE PATTERN DOTS-2456
+<U282D>     /x58         BRAILLE PATTERN DOTS-1346
+<U283D>     /x59         BRAILLE PATTERN DOTS-13456
+<U2835>     /x5a         BRAILLE PATTERN DOTS-1356
+<U282A>     /x5b         BRAILLE PATTERN DOTS-246
+<U2833>     /x5c         BRAILLE PATTERN DOTS-1256
+<U283B>     /x5d         BRAILLE PATTERN DOTS-12456
+<U2818>     /x5e         BRAILLE PATTERN DOTS-45
+<U2838>     /x5f         BRAILLE PATTERN DOTS-456
+END CHARMAP
+
+WIDTH
+<U2800>...<U283F>	1
+END WIDTH
